CI fails with experimental: ValueError: Pointer argument (at 0) cannot be accessed from Triton (cpu tensor?)

CI fails with experimental: https://github.com/huggingface/trl/actions/runs/20947812023/job/60194210545
> ValueError: Pointer argument (at 0) cannot be accessed from Triton (cpu tensor?)
```python
  FAILED tests/experimental/test_winrate_callback.py::TestWinRateCallback::test_soft_judge - ValueError: Pointer argument (at 0) cannot be accessed from Triton (cpu tensor?)
```

Stacktrace:
```python
   >       trainer.train()
  
  tests/experimental/test_winrate_callback.py:157: 
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  .venv/lib/python3.13/site-packages/transformers/trainer.py:2325: in train
      return inner_training_loop(
  .venv/lib/python3.13/site-packages/transformers/trainer.py:2573: in _inner_training_loop
      self.control = self.callback_handler.on_train_begin(args, self.state, self.control)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.13/site-packages/transformers/trainer_callback.py:506: in on_train_begin
      return self.call_event("on_train_begin", args, state, control)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.13/site-packages/transformers/trainer_callback.py:556: in call_event
      result = getattr(callback, event)(
  trl/experimental/winrate_callback.py:175: in on_train_begin
      self.ref_completions = _generate_completions(
  trl/experimental/winrate_callback.py:70: in _generate_completions
      generations = unwrapped_model.generate(
  .venv/lib/python3.13/site-packages/torch/utils/_contextlib.py:120: in decorate_context
      return func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.13/site-packages/transformers/generation/utils.py:2564: in generate
      result = decoding_method(
  .venv/lib/python3.13/site-packages/transformers/generation/utils.py:2784: in _sample
      outputs = self(**model_inputs, return_dict=True)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.13/site-packages/torch/nn/modules/module.py:1775: in _wrapped_call_impl
      return self._call_impl(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.13/site-packages/torch/nn/modules/module.py:1786: in _call_impl
      return forward_call(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.13/site-packages/transformers/utils/generic.py:918: in wrapper
      output = func(self, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.13/site-packages/transformers/models/qwen2/modeling_qwen2.py:449: in forward
      outputs: BaseModelOutputWithPast = self.model(
  .venv/lib/python3.13/site-packages/torch/nn/modules/module.py:1775: in _wrapped_call_impl
      return self._call_impl(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.13/site-packages/torch/nn/modules/module.py:1786: in _call_impl
      return forward_call(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.13/site-packages/transformers/utils/generic.py:1072: in wrapper
      outputs = func(self, *args, **kwargs)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.13/site-packages/transformers/models/qwen2/modeling_qwen2.py:384: in forward
      hidden_states = decoder_layer(
  .venv/lib/python3.13/site-packages/transformers/modeling_layers.py:94: in __call__
      return super().__call__(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.13/site-packages/torch/nn/modules/module.py:1775: in _wrapped_call_impl
      return self._call_impl(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.13/site-packages/torch/nn/modules/module.py:1786: in _call_impl
      return forward_call(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.13/site-packages/transformers/utils/deprecation.py:172: in wrapped_func
      return func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.13/site-packages/transformers/models/qwen2/modeling_qwen2.py:232: in forward
      hidden_states = self.input_layernorm(hidden_states)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.13/site-packages/torch/nn/modules/module.py:1775: in _wrapped_call_impl
      return self._call_impl(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.13/site-packages/torch/nn/modules/module.py:1786: in _call_impl
      return forward_call(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.13/site-packages/liger_kernel/transformers/rms_norm.py:33: in forward
      return LigerRMSNormFunction.apply(
  .venv/lib/python3.13/site-packages/torch/autograd/function.py:581: in apply
      return super().apply(*args, **kwargs)  # type: ignore[misc]
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.13/site-packages/liger_kernel/ops/utils.py:40: in wrapper
      return fn(ctx, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.13/site-packages/liger_kernel/ops/rms_norm.py:556: in forward
      Y, X, RSTD, BLOCK_SIZE, num_warps, casting_mode = rms_norm_forward(X, W, eps, offset, casting_mode, row_mode)
                                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.13/site-packages/liger_kernel/ops/rms_norm.py:400: in rms_norm_forward
      _rms_norm_forward_kernel[(n_rows,)](
  .venv/lib/python3.13/site-packages/triton/runtime/jit.py:419: in <lambda>
      return lambda *args, **kwargs: self.run(grid=grid, warmup=False, *args, **kwargs)
                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.13/site-packages/triton/runtime/jit.py:757: in run
      kernel.run(grid_0, grid_1, grid_2, stream, kernel.function, kernel.packed_metadata, launch_metadata,
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  
  self = <triton.backends.nvidia.driver.CudaLauncher object at 0x7fdfb84eaba0>
  gridX = 14, gridY = 1, gridZ = 1, stream = 0, function = 94058060627088
  args = ((4, 1, 0, 1, 1, 1), <triton.compiler.compiler.LazyDict object at 0x7fe05fae1070>, <triton.knobs.HookChain object at 0...536e-43,  0.0000e+00,  2.0179e-43,  0.0000e+00,  4.8631e-33,
            1.5105e-36, -5.6743e-05,  3.0687e-41]]), 8, ...)
  allocate_scratch = <function CudaLauncher.__call__.<locals>.allocate_scratch at 0x7fe05fa58180>
  global_scratch = None, profile_scratch = None
  
      def __call__(self, gridX, gridY, gridZ, stream, function, *args):
      
          def allocate_scratch(size, align, allocator):
              if size > 0:
                  grid_size = gridX * gridY * gridZ
                  alloc_size = grid_size * self.num_ctas * size
                  alloc_fn = allocator.get()
                  return alloc_fn(alloc_size, align, stream)
              return None
      
          global_scratch = allocate_scratch(self.global_scratch_size, self.global_scratch_align, _allocation._allocator)
          profile_scratch = allocate_scratch(self.profile_scratch_size, self.profile_scratch_align,
                                             _allocation._profile_allocator)
  >       self.launch(gridX, gridY, gridZ, stream, function, self.launch_cooperative_grid, self.launch_pdl,
                      global_scratch, profile_scratch, *args)
  E       ValueError: Pointer argument (at 0) cannot be accessed from Triton (cpu tensor?)
  
  .venv/lib/python3.13/site-packages/triton/backends/nvidia/driver.py:712: ValueError
```

--- a/tests/experimental/test_winrate_callback.py
+++ b/tests/experimental/test_winrate_callback.py
@@ -50,10 +50,7 @@ class TrainerWithRefModel(Trainer):
             processing_class=processing_class,
         )
         # Prepare ref_model like TRL trainers do (DPOTrainer, GRPOTrainer, etc.)
-        if ref_model is not None:
-            self.ref_model = self.accelerator.prepare_model(ref_model, evaluation_mode=True)
-        else:
-            self.ref_model = None
+        self.ref_model = self.accelerator.prepare_model(ref_model, evaluation_mode=True)
 
 
 class TestWinRateCallback(TrlTestCase):

